PubNub/Network/Parsers/PNSubscribeParser.m

```diff
@@ -188,36 +177,27 @@ + (NSDictionary *)parsedServiceResponse:(id)response withData:(NSDictionary *)ad
         }
         processedResponse = @{@"events":feedEvents,@"timetoken":timeToken};
     }
-    
+
     return processedResponse;
 }
 
 
 #pragma mark - Events processing
 
-+ (NSMutableDictionary *)eventFromData:(id)data forChannel:(NSString *)channel
-                                 group:(NSString *)group
-              withAdditionalParserData:(NSDictionary *)additionalData {
++ (NSMutableDictionary *)eventFromData:(id)data withAdditionalParserData:(NSDictionary *)additionalData {
     
     NSMutableDictionary *event = [NSMutableDictionary new];
-    if ([channel length]) {
-        
-        event[(![group length] ? @"subscribedChannel": @"actualChannel")] = channel;
-    }
-    if ([group length]) {
-        
-        event[@"subscribedChannel"] = group;
-    }
-    
+    NSString *channel = data[kPNMessageChannelKey];
+    NSString *subscriptionMatch = data[kPNMessageSubscriptionMatchKey];
+    event[@"subscribedChannel"] = subscriptionMatch;
+    event[@"actualChannel"] = channel;
     BOOL isPresenceEvent = [PNChannel isPresenceObject:channel];
-    if (![channel length] && [data isKindOfClass:[NSDictionary class]]) {
-        
-        isPresenceEvent = (data[@"timestamp"] != nil &&
-                           (data[@"action"] != nil || data[@"occupancy"] != nil));
+    if ([data[@"d"] isKindOfClass:[NSDictionary class]]) {
+        NSDictionary *payload = data[@"d"];
+        isPresenceEvent = (payload[@"timestamp"] != nil &&
+                           (payload[@"action"] != nil || payload[@"occupancy"] != nil));
     }
```

Comment:
@jzucker2 because of note above, I suggest to remove this checks, because passed channel name allow us to find out whether this is presence event (arrived on presence channel) or not.